librz/analysis/fcn.c

```diff
@@ -650,6 +647,7 @@ static RzAnalysisBBEndCause run_basic_block_analysis(RzAnalysisTaskItem *item, R
 		// we checked before whether there is a bb at addr, so the create should have succeeded
 		rz_return_val_if_fail(bb, RZ_ANALYSIS_RET_ERROR);
 	}
+	bb->sp_entry = sp;
```

Comment:
Probably worth adding a comment here like `// Initialize the initial SP value for the basic block` since it might be not clear for someone new who would need to dive in this code.